pkg/plugins/gateway/algorithms/least_busy_time.go

```diff
@@ -51,15 +49,11 @@ func NewLeastBusyTimeRouter() (types.Router, error) {
 	}, nil
 }
 
-func (r leastBusyTimeRouter) Route(ctx *types.RoutingContext, pods types.PodList) (string, error) {
+func (r leastBusyTimeRouter) Route(ctx *types.RoutingContext, readyPodList types.PodList) (string, error) {
 	var targetPod *v1.Pod
 	minBusyTimeRatio := math.MaxFloat64 // <= 1 in general
 
-	if pods.Len() == 0 {
-		return "", fmt.Errorf("no available pods for request routing")
-	}
-
-	for _, pod := range pods.All() {
+	for _, pod := range readyPodList.All() {
```

Comment:
ip check at line 57 can be removed. FilterRoutablePods already does it. Please remove this at other places if present.